src/packaging/metadata.py

```diff
@@ -761,62 +756,66 @@ def from_email(
     *validate* parameter)"""
     version: _Validator[version_module.Version] = _Validator()
     """:external:ref:`core-metadata-version` (required)"""
-    dynamic: _Validator[List[str]] = _Validator(
+    dynamic: _Validator[Union[List[str], None]] = _Validator(
```

Comment:
```suggestion
    dynamic: _Validator[Optional[List[str]]] = _Validator(
```

There's a dedicated type for unions with None, which I find slightly easier to read! How do you feel about changing over?

Reply:
I don't have a strong opinion, I'm happy to switch. I only used Union because you eventually get the `T | None` syntax.